src/main/kotlin/nomic/data/repositories/games/FilterGamesByUserSpecification.kt

```diff
@@ -0,0 +1,20 @@
+package nomic.data.repositories.games
+
+import nomic.data.dtos.GameDTO
+import nomic.data.dtos.Games
+import nomic.domain.entities.EndUser
+import org.ktorm.dsl.eq
+import org.ktorm.entity.EntitySequence
+import org.ktorm.entity.filter
+
+/**
+ * This specification filters games from the repository call to contain only games created
+ * by the provider user.
+ *
+ * @see[user] The user to filter the games by;
+ */
+class FilterGamesByUserSpecification(private val user: EndUser) : IGameSpecification {
+    override fun apply(sequence: EntitySequence<GameDTO, Games>): EntitySequence<GameDTO, Games> {
+        return sequence.filter { user.id eq Games.userId }
+    }
+}
```

Comment:
Still confused on the actual purpose of this it would be nice to have a through explanation on the actual issue/problem you are trying to solve with this file.